### ISSUE TITLE
differential assets loading

### ISSUE THREAD

**Issue by digitall-it** [silverstripe/sspak#48]
Feature request: ability to only perform differential uploads and downloads via rsync.  This would speed up asset syncing if done frequently. We all understand that even if this is not an edge use case, and that better suited tools exists for that scope (like Deployer), it would be nice and very useful to incorporate this functionality.

**Comment by dhensby** [comment]
Why not just use `rsync`?

**Comment by digitall-it** [comment]
sspak also syncs the database

**Comment by dhensby** [comment]
you can use sspak to do the DB (that won't be incremental) and rsync for the assets?

**Comment by digitall-it** [comment]
Yes, but probably that would be better suited as a proper deployer recipe. 

Sspak is a good basic entry point to automated deployment and I've found out it works well if coupled with [Plesk Gitman](https://devblog.plesk.com/2015/12/gitman/), if only it could do differential assets loading. Plesk can even detect Composer repos and update/install them as per instructions. 

It could smooth out transition to something more fancy like Capistrano, Deployer, Rocketeer, that also enable to flush and dev/build. 

But as more I write this answer, the more I understand that this could open up to more scope creeping that this little gem should handle on its own shoulders alone.

**Comment by dhensby** [comment]
That's my thinking really. sspak has a very narrow scope and being a wrapper for `rsync` calls is not really one of them...
